descwl/analysis.py

```diff
@@ -1046,123 +1106,134 @@ def finalize(self,verbose,trace,calculate_bias,no_analysis):
+                        data['ds'][galaxy] = np.sqrt(iso_variance[ds_index])
+                        data['dg1'][galaxy] = np.sqrt(iso_variance[dg1_index])
+                        data['dg2'][galaxy] = np.sqrt(iso_variance[dg2_index])
+                        data['cond_num'][galaxy] = cond_num
+
+                        if self.calculate_bias:
+                            iso_bias = results.get_bias([galaxy], iso_covariance.copy())
+                            data['bias_f'][galaxy] = iso_bias[dflux_index]
+                            data['bias_s'][galaxy] = iso_bias[ds_index]
+                            data['bias_g1'][galaxy] = iso_bias[dg1_index]
+                            data['bias_g2'][galaxy] = iso_bias[dg2_index]
+                            data['bias_x'][galaxy] = iso_bias[dx_index]
+                            data['bias_y'][galaxy] = iso_bias[dy_index]
+
+            # Order group members by decreasing isolated S/N (if available), otherwise use snr_sky.
+            #this assumes that snr_sky is close to snr_iso (although not necessarily the same.)
+            if not self.no_fisher:
+                sorted_indices = group_indices[np.argsort(data['snr_iso'][grp_members])[::-1]]
+            else: 
+                sorted_indices = group_indices[np.argsort(data['snr_sky'][grp_members])[::-1]]
+
             data['grp_rank'][sorted_indices] = np.arange(grp_size,dtype = np.int16)
             # Replace group ID with ID of galaxy with largest S/N.
             group_leader = data['db_id'][sorted_indices[0]]
             data['grp_id'][grp_members] = group_leader
 
-            alpha = self.alpha
-            detection_threshold = 6. # cut on snr_grpf
-            data['g1_fit'][sorted_indices] = 0.
-            data['g2_fit'][sorted_indices] = 0.
-            detected = (data['snr_grpf'][sorted_indices] > detection_threshold)
-            if np.count_nonzero(detected) > 0 and grp_size > 1 and not self.no_lmfit:
-                use_count = np.zeros(grp_size,dtype = int)
-                # Loop over galaxies in order of decreasing snr_iso.
-                for i1,g1 in enumerate(sorted_indices):
-                    # Skip sources below our detection threshold, which instead are added
-                    # to a detected object below (unless they only directly overlap other
-                    # undetected sources).
-                    if not detected[i1]:
-                        continue
-                    stamp1 = results.get_stamp(g1)
-                    deblended = stamp1.copy()
-                    # Loop over other galaxies in this group in order of decreasing snr_iso.
-                    for i2,g2 in enumerate(sorted_indices):
-                        if i1 == i2 or not overlapping_bounds[g1,g2]:
+            if not self.no_fisher and not self.no_lmfit:
+                alpha = self.alpha
+                detection_threshold = 6. # cut on snr_grpf
```

Comment:
hmmmm, I would suggest to do this after running or add it as a configurable parameter and not hardcode it here.

Reply:
@fjaviersanchez do you have an obvious candidate for what the (default) detection threshold of what each survey should be ? If so I can do that and otherwise use 6 as a default value throughout for now.

Reply:
It's 5 for DES and HSC (it's in the overleaf draft in page 28)

Reply:
I'm resolving this for now since I addressed this with the new `--detection-threshold` command line argument.